### ISSUE TITLE
Why build_dataset.py combine source = "instruction + input + output" ==> overfit immediately ?

### ISSUE THREAD

**Issue by thusinh1969** [ymcui/Chinese-LLaMA-Alpaca#749]
### Check before submitting issues

- [X] Make sure to pull the latest code, as some issues and bugs have been fixed.
- [X] Due to frequent dependency updates, please ensure you have followed the steps in our [Wiki](https://github.com/ymcui/Chinese-LLaMA-Alpaca/wiki)
- [X] I have read the [FAQ section](https://github.com/ymcui/Chinese-LLaMA-Alpaca/wiki/FAQ) AND searched for similar issues and did not find a similar problem or solution
- [X] Third-party plugin issues - e.g., [llama.cpp](https://github.com/ggerganov/llama.cpp), [text-generation-webui](https://github.com/oobabooga/text-generation-webui), [LlamaChat](https://github.com/alexrozanski/LlamaChat), we recommend checking the corresponding project for solutions
- [X] Model validity check - Be sure to check the model's [SHA256.md](https://github.com/ymcui/Chinese-LLaMA-Alpaca/blob/main/SHA256.md). If the model is incorrect, we cannot guarantee its performance

### Type of Issue

Model training and fine-tuning

### Base Model

LLaMA-7B

### Operating System

Linux

### Describe your issue in detail

Repo original file:  build_dataset.py, row 47 - 48:
...
        for s,t in zip(tokenized_sources['input_ids'],tokenized_targets['input_ids']):
            input_ids = torch.LongTensor(s + t)[:max_seq_length]
...

Why do you combine t + s (which is target) into source ? Would this overfit the model almost immediately ?

Thanks,
Steve

### Dependencies (must be provided for code-related issues)

Original repo "build_dataset.py"

### Execution logs or screenshots

Nothing, just curious about repo code.

**Comment by airaria** [comment]
I'm sorry but I didn't get your point. Why do you think the model will be overfitting?
The training loss is the standard CLM loss except that it is only calculated on the output:
https://github.com/ymcui/Chinese-LLaMA-Alpaca/blob/01eacec75fbd93c4b472aa8c2f9d7116607e17a4/scripts/training/build_dataset.py#L45-L47

**Comment by thusinh1969** [comment]
input_ids = torch.LongTensor(s + t)[:max_seq_length] 

This line said: you combined INSTRUCT + INPUT + TARGET as overall INPUT. Why ?

Steve
